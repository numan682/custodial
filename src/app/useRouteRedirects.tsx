import { useEffect } from 'react'
import { useSelector } from 'react-redux'
import { useNavigate } from 'react-router-dom'

import { selectAddress } from 'app/state/wallet/selectors'

export const useRouteRedirects = () => {
  const address = useSelector(selectAddress)
  const navigate = useNavigate()

  useEffect(() => {
    if (address) {
      navigate(`/account/${address}`)
    }
  }, [address, navigate])
}
