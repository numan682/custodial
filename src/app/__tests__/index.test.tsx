import * as React from 'react'
import { createRenderer } from 'react-test-renderer/shallow'

import { App } from '../index'
import { useRouteRedirects } from '../useRouteRedirects'

const renderer = createRenderer()

jest.mock('../useRouteRedirects')

describe('<App />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<App />)

    const renderedOutput = renderer.getRenderOutput()
    expect(useRouteRedirects).toHaveBeenCalled()
    expect(renderedOutput).toMatchSnapshot()
  })
})
