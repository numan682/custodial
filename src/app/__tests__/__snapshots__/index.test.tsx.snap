// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render and match the snapshot 1`] = `
<FatalErrorHandler>
  <ModalProvider>
    <Helmet
      defaultTitle="Oasis Wallet"
      defer={true}
      encodeSpecialCharacters={true}
      htmlAttributes={
        {
          "lang": "en-US",
        }
      }
      prioritizeSeoTags={false}
      titleTemplate="%s - Oasis Wallet"
    >
      <meta
        content="A wallet for Oasis"
        name="description"
      />
    </Helmet>
    <BuildBanner />
    <Box
      background="background-back"
      direction="row-responsive"
      fill={true}
      style={
        {
          "minHeight": "100vh",
        }
      }
    >
      <PersistLoadingGate>
        <UnlockGate>
          <Navigation />
          <Box
            flex={true}
            pad={
              {
                "top": undefined,
              }
            }
          >
            <Main>
              <Toolbar />
              <Outlet />
              <Memo />
            </Main>
          </Box>
        </UnlockGate>
      </PersistLoadingGate>
    </Box>
  </ModalProvider>
</FatalErrorHandler>
`;
